src/libutil/tarfile.cc

```diff
@@ -81,6 +81,7 @@ TarArchive::TarArchive(Source & source, bool raw, std::optional<std::string> com
     if (!raw) {
         archive_read_support_format_tar(archive);
         archive_read_support_format_zip(archive);
+        archive_read_support_format_empty(archive);
```

Comment:
Is this for uncompressed tarballs?
Could we add a test for that?

Reply:
No, it's for empty files.

Reply:
This is very surprising, maybe add a few sentences?